### PR TITLE
unar: fix build on Linux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/unar.rb [hunk 1]
```diff
@@ -17,6 +17,16 @@ class Unar < Formula
 
   depends_on xcode: :build
 
+  uses_from_macos "llvm" => [:build, :test]
+
```
Comment (alebcay) [inline Formula/unar.rb:21]: Based on linkage test for Linux, looks like might want `uses_from_macos "bzip2"`.

**Comment by danielnachun** [comment]
> Thanks @danielnachun. One thing that I noticed while building this in container is that all of the Makefile invocations still seem to be calling `gcc` (despite the `fails_with :gcc` line). The build seems to work fine though. Is this something we want to look into further?

This is okay because the compiler shim "intercepts" these calls and reroutes them to `clang`.  If the Makefile were hardcoding a path to `gcc` it would be a problem, but our compiler shims always come first in PATH so that our `gcc` (which is just a symlink to the shim) is used instead.

> I also see there's linkage to some indirect dependencies, e.g. `zlib`, `icu4c` and `libobjc2`. I think it would be good to `uses_from_macos` those.

I think we don't audit for indirect linkage because it would lead to even larger dependency trees than we already have, especially for `icu4c` which is already very difficult to update as is.  This is a small formula so I'm not strongly opposed to adding them, but I think we have to conservative about this in general to keep dependency trees from ballooning too much.

**Comment by alebcay** [comment]
> I think we don't audit for indirect linkage because it would lead to even larger dependency trees than we already have, especially for `icu4c` which is already very difficult to update as is. This is a small formula so I'm not strongly opposed to adding them, but I think we have to be conservative about this in general to keep dependency trees from ballooning too much.

I'm fine with the reasoning here specifically with regard to `unar` but I think moving forward it may be helpful to see if there are any situations where we should be more strict with catching direct linkage, or if there any situations where we may develop testing "blind spots" that may result from applying this approach more broadly.

**Comment by danielnachun** [comment]
> I'm fine with the reasoning here specifically with regard to `unar` but I think moving forward it may be helpful to see if there are any situations where we should be more strict with catching direct linkage, or if there any situations where we may develop testing "blind spots" that may result from applying this approach more broadly.

This is definitely tricky to find the right balance.  Normally we should be covered by dependent testing when we revision bump direct dependents of a library, but I can't say for sure if that rules out edge cases where a formula doesn't get revision bumped when it should be.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**